pkg/controller/build/build_controller.go

```diff
@@ -792,139 +811,115 @@ func (ctrl *Controller) postBuildCleanup(pool *mcfgv1.MachineConfigPool, ignoreM
+				Status: metav1.ConditionFalse,
 			},
 			{
-				Type:   mcfgv1.MachineConfigPoolBuildSuccess,
+				Type:   string(mcfgv1alpha1.MachineOSReady),
 				Reason: "BuildSucceeded",
-				Status: corev1.ConditionTrue,
+				Status: metav1.ConditionTrue,
 			},
 			{
-				Type:   mcfgv1.MachineConfigPoolBuilding,
-				Status: corev1.ConditionFalse,
+				Type:   string(mcfgv1alpha1.MachineOSBuilding),
+				Status: metav1.ConditionFalse,
 			},
 			{
-				Type:   mcfgv1.MachineConfigPoolDegraded,
-				Status: corev1.ConditionFalse,
+				Type:   string(mcfgv1.MachineConfigPoolDegraded),
+				Status: metav1.ConditionFalse,
 			},
 		})
 
-		return ctrl.updatePoolAndSyncAvailableStatus(ps.MachineConfigPool())
+		return ctrl.updateMOSBAndSyncAvailable(bs.Build)
+
+		// don't SetImagPullSpec. It is already in MOSB. Wherever we check for desiredImage or query the Pool's annotation
+		// we need to replace with mosb
 	})
 }
 
 // Marks a given MachineConfigPool as build pending.
-func (ctrl *Controller) markBuildPendingWithObjectRef(ps *poolState, objRef corev1.ObjectReference) error {
+func (ctrl *Controller) markBuildPendingWithObjectRef(mosb *mcfgv1alpha1.MachineOSBuild, objRef corev1.ObjectReference) error {
 	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		mcp, err := ctrl.mcfgclient.MachineconfigurationV1().MachineConfigPools().Get(context.TODO(), ps.Name(), metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-
-		ps := newPoolState(mcp)
-
-		klog.Infof("Build for %s marked pending with object reference %v", ps.Name(), objRef)
+		bs := ctrlcommon.NewMachineOSBuildState(mosb)
 
-		ps.SetBuildConditions([]mcfgv1.MachineConfigPoolCondition{
+		bs.SetBuildConditions([]metav1.Condition{
 			{
-				Type:   mcfgv1.MachineConfigPoolBuildInterrupted,
-				Status: corev1.ConditionFalse,
+				Type:   string(mcfgv1alpha1.MachineOSBuildInterrupted),
+				Status: metav1.ConditionFalse,
 			},
 			{
-				Type:   mcfgv1.MachineConfigPoolBuildFailed,
-				Status: corev1.ConditionFalse,
+				Type:   string(mcfgv1.MachineConfigPoolBuildFailed),
+				Status: metav1.ConditionFalse,
 			},
 			{
-				Type:   mcfgv1.MachineConfigPoolBuildSuccess,
-				Status: corev1.ConditionFalse,
+				Type:   string(mcfgv1alpha1.MachineOSReady),
+				Status: metav1.ConditionFalse,
 			},
+			// what is the difference between building and build pending?
 			{
-				Type:   mcfgv1.MachineConfigPoolBuilding,
-				Status: corev1.ConditionFalse,
-			},
-			{
-				Type:   mcfgv1.MachineConfigPoolBuildPending,
+				Type:   string(mcfgv1alpha1.MachineOSBuilding),
 				Reason: "BuildPending",
-				Status: corev1.ConditionTrue,
+				Status: metav1.ConditionTrue,
 			},
+			/*
+				{
+					Type:   mcfgv1.MachineConfigPoolBuildPending,
+					Status: metav1.ConditionTrue,
+				},
+			*/
 		})
 
-		// If the MachineConfigPool has the build object reference, we just want to
-		// update the MachineConfigPool's status.
-		if ps.HasBuildObjectRef(objRef) {
```

Comment:
thought: This never worked quite the way that I wanted it to only because RenderController would wipe out the object reference. In the long-run, I'd like to move away from this as the way that we're associating a running build with a MachineConfigPool. Instead, I think adding labels to the MachineConfigPool that we can use label queries for might be the better option.

(To be clear: I'm not asking you to change anything; I'm commenting on something that I think may have been a mistake on my part.)